autorest/azure/token.go

```diff
@@ -104,6 +154,79 @@ func (spt *ServicePrincipalToken) EnsureFresh() error {
 
 // Refresh obtains a fresh token for the Service Principal.
 func (spt *ServicePrincipalToken) Refresh() error {
+	newToken, err := spt.GetOauthToken()
+	if err != nil {
+		return err
+	}
+	spt.Token = *newToken
+
+	return nil
+}
+
+// SetAuthValues is a method of the interface ServicePrincipalTokenSecret
+// It will populate the form submitted during oAuth Token Acquisition using a static client_secret
+func (tokenSecret *ServicePrincipalTokenSecret) SetAuthValues(spt *ServicePrincipalToken, v *url.Values) error {
+	v.Set("client_secret", tokenSecret.ClientSecret)
+	return nil
+}
+
+// SignedJwt returns the JWT signed with the secret's certData
+func (secret *ServicePrincipalCertSecret) SignedJwt(spt *ServicePrincipalToken) (string, error) {
+	privateKey, cert, err := pkcs12.Decode(secret.Pkcs12Bytes, secret.Password)
+	if err != nil {
+		return "", err
+	}
+
+	rsaPrivKey, isRsaKey := privateKey.(*rsa.PrivateKey)
+	if !isRsaKey {
+		return "", fmt.Errorf("PKCS12 certificate must contain an RSA private key")
+	}
+
+	hasher := sha1.New()
```

Comment:
Could these two statements be combined (e.g., sha1.New().Write(cert.Raw)). Since sha1.New does not return an error, introducing an unnecessary variable, to me, clutters things.